### ISSUE TITLE
Generate colmap data

### ISSUE THREAD

**Issue by massyzs** [dunbar12138/DSNeRF#102]
I try to generate data like yours. But cannot. I only got 400-500 3D points in total instead of more than 2000. No matter by GUI or Command-Line mode. And it seems that you did not provide how to get the npy file from output.

**Comment by massyzs** [comment]
10 images will get me more than 2000 depth points. I strongly need an explanation on how to get 2000 depth points by two images.

**Comment by massyzs** [comment]
I even used the original resolution of two images. (4032*3024, yours is 1008*756). The colmap output 900 3D points which is impossible to get 2000 depth information. I think you should clarify this problem as soon as possible. Or I think it's necessary to report this to the CVPR conference.

**Comment by dunbar12138** [comment]
Hi, sorry for the confusion. We run COLMAP on images of the **original** resolution and downscale the cameras and depths when post-processing the data [here](https://github.com/dunbar12138/DSNeRF/blob/7d070be4d067ef2df3e2c75df4680818a4c72544/load_llff.py#L107) and [here](https://github.com/dunbar12138/DSNeRF/blob/7d070be4d067ef2df3e2c75df4680818a4c72544/load_llff.py#L371).

Here is an example COLMAP run to get over 2000 keypoints on two views of the fern scene.

```
colmap feature_extractor --database_path database.db --image_path images  --SiftExtraction.max_image_size 4032 --SiftExtraction.max_num_features 32768 --SiftExtraction.estimate_affine_shape 1 --SiftExtraction.domain_size_pooling 1

colmap exhaustive_matcher --database_path database.db --SiftMatching.guided_matching 1 --SiftMatching.max_num_matches 65536

mkdir sparse

colmap mapper --database database.db --image_path images --output_path sparse --Mapper.ba_local_max_num_iterations 40 --Mapper.ba_local_max_refinements 3 --Mapper.ba_global_max_num_i
terations 100
```
Here is link to the COLMAP result I got with 2604 keypoints: https://drive.google.com/file/d/1YqozhVM63u9aN1r2ibW5Gqlm-i3iZad2/view?usp=sharing

Actually, I used to be able to get such many keypoints by using the default extreme quality setting of COLMAP, e.g. `colmap automatic_reconstructor --image_path images --workspace_path ./ --sparse 1 --dense 0 --quality extreme`. But colmap seems to have updated the settings. So now I have to specify the arguments for each stage. Actually, most of the settings are still the same as extreme quality [here](https://github.com/colmap/colmap/blob/c04629017e7378b3046c6e8961277fbe98b56a32/src/colmap/controllers/option_manager.cc#L165) except that I increase the max number of features and image size. This setting might still not perfectly match what's in the dataset, but you can always further increase the max number of features and matches to get more keypoints.

Thanks for pointing this out. I'll update the document accordingly.

**Comment by massyzs** [comment]
Yes, thanks for the clarification. The imgs2poses.py file indeed indicates the wrong command which will make people doubt the experiment result. I will try the auto-one with "--quality extreme", and your step-by-step command.

**Comment by massyzs** [comment]
I've tried. Indeed the Colmap changed the default setting and your command could help to generate more than 2000 depth points. Thanks
